src/util/fields.py

```diff
@@ -1585,3 +1585,59 @@ def update_server_actions_fields(cr, src_model, dst_model=None, fields_mapping=N
         ) % {"src_model": src_model, "dst_model": dst_model, "actions": ", ".join(action_names)}
 
         add_to_migration_reports(message=msg, category="Server Actions")
+
+
+def convert_m2o_to_m2m(cr, origin_model, m2o_field, m2m_field):
+    """
+    TODO
+    """
+    _validate_model(origin_model)
+
+    if not table_exists(cr, m2m_field):
+        raise SleepyDeveloperError("m2m table should already exist warning")
```

Comment:
IMO this util should create the m2m the table (auto key?). 

Still I don't get what you are checking here. Why should there be a table with the name of the m2m_field? Maybe you wanted to check a _column_?

If we auto-create the m2m table, we will allow simpler code for fields that are introduced as m2o and later changed to m2m in the **same** major version --think 18.2 and 18.4 for example.